### ISSUE TITLE
Add Rutube service to NewPipe

### ISSUE THREAD

**Issue by Nightscreamer** [TeamNewPipe/NewPipeExtractor#827]
Add, please, Rutube service to NewPipe.
It could be useful:
https://github.com/TalbotGooday/Android-Oembed-Video

**Comment by basiliscos** [comment]
please, add ASAP, since there are rumors, that youtube is going to be blocked in Russia.

**Comment by em92** [comment]
Rutube becomes popular in Russia since YouTube started blocking some accounts from Russia.

Recently implemented way to retrieve list of videos from channel and playlist https://github.com/RSS-Bridge/rss-bridge/pull/2661

**Comment by makhov** [comment]
My 2 cents on why it's not a good idea:

Rutube is a part of Gazprom Media – a media holding owned by Gazprom (>50% of shares belong to the Russian state). Of course, it's highly connected with the Russian government. 
It also owns the NTV channel which is one of the most popular channels in Russia that produces propaganda and supports the Ukraine invasion. 

The current Gazprom Media CEO Alexandr Zharov was a head of Roskomnadzor in 2012-2020. 
He is the one, who was creating the censorship machine in Russia. Alexandr Zharov is on the sanctions list as a Russian official responsible for the war.
Currently in Russia:
- banned all independent Russian media
- banned all foreign media that write in Russian
- banned Twitter
- banned Facebook
- banned Instagram

RKN also tests youtube ban from time to time, but they are not yet ready (technically or politically) to do it. 
That's why helping them and giving one more reason to ban YouTube is a bad idea.

**Comment by Nightscreamer** [comment]
> My 2 cents on why it's not a good idea:
[...]
> RKN also tests youtube ban from time to time, but they are not yet ready (technically or politically) to do it. That's why helping them and giving one more reason to ban YouTube is a bad idea.

First of all, this is not the place for politics. But now that you're talking about it, tell us about Google, Twitter, Facebook, etc. Are you sure they have no ties to the U.S. government?

So let's leave politics to the politicians and get down to more useful things for the people =)

**Comment by DI555** [comment]
yes , good idea, - to support rutube video , and also online streamings 👍 .
no politics , just usability !!!

**Comment by makhov** [comment]
It is a place for politics, like any other place where problems related to human life and society are discussed.

Of course, it's up to you, but I can assure you that helping RKN build an iron curtain and break the Internet in Russia is not a good idea in general. I know what I speak about.

And it's even more stupid if you're in Russia, it's like digging your own [professional] grave.

**Comment by em92** [comment]
@makhov, YouTube's censorship actions increased demand (спрос) of using Rutube in Russia for uploading videos and watching them. 

Some users use official Rutube Android app to watch videos. You sentence about "don't implement it" means that, those users continue using that official app. Are you ok with that?

**Comment by makhov** [comment]
> Some users use official Rutube Android app to watch videos. You sentence about "don't implement it" means that, those users continue using that official app. Are you ok with that?

@em92 Good point. What I'm not ok with is the spread of propaganda. So there are three options for what will happen  after adding the feature:
- People use NewPipe instead of the official app and have access to other sources of information (which is good)
- People who use NewPipe get access to propaganda (which is bad)
- Nothing changes

Considering audiences of NewPipe and Rutube app, I bet on the two latter options.

**Comment by czarkoff** [comment]
> What I'm not ok with is the spread of propaganda.

Software shouldn't make moral judgement calls.  It should facilitate users' choices, not gatekeep.

**Comment by kuhtinov** [comment]
Everyone is good at politics. But java coders, as we can see, are not among us.

**Comment by makhov** [comment]
> Software shouldn't make moral judgement calls. It should facilitate users' choices, not gatekeep.

@czarkoff Another good point, thanks! That's true, the software doesn't make such decisions, but people (and developers among them) do. And people do it based on their own principles, experience, and point of view.

**Comment by DI555** [comment]
i suppose, many alrernative channels (english, french and spanish speaking) which was banned on yt, will got their audience again with this feature !!!

**Comment by bubbec** [comment]
-1

I completely agree with these guys @makhov @AIGeneratedUsername.
